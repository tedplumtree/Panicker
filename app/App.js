import React from 'react';
import { ThemeProvider } from 'react-native-material-ui';
import { Provider as StoreProvider } from 'react-redux';
import { createStore } from 'redux';
import { MaterialDesignTheme } from './config/styles';
import SettingsAndActivateScreen from './screens/SettingsAndActivateScreen';
import combinedReducers from './reducers/index';

const store = createStore(combinedReducers);

const App = () => (
  <StoreProvider store={store}>
    <ThemeProvider uiTheme={MaterialDesignTheme}>
      <SettingsAndActivateScreen />
    </ThemeProvider>
  </StoreProvider>
);

export default App;
