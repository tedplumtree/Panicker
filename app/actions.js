export const TOGGLE_SENSOR = 'TOGGLE_SENSOR';

export const toggleSensor = () => ({
  type: TOGGLE_SENSOR,
});
