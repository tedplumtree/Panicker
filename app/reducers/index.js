import { combineReducers } from 'redux';
import sensor from './sensor';

const combined = combineReducers({
  sensor,
});

export default combined;
