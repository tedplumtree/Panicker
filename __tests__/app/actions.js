import { toggleSensor } from '../../app/actions';

describe('toggle sensor action', () => {
  it('generates the action', () => {
    expect(toggleSensor()).toMatchSnapshot();
  });
});
