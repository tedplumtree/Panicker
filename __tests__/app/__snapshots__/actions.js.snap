// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toggle sensor action generates the action 1`] = `
Object {
  "type": "TOGGLE_SENSOR",
}
`;
